File: tools/wave/src/data/route.js
const ALL = "all";
const GET = "get";
const POST = "post";
const PUT = "put";
const DELETE = "delete";
const STATIC = "static";
const OPTIONS = "options";

/**
 * @module Route
 */
class Route {
  constructor({method = ALL, uri = "/", handler = () => {}, directory = "/"} = {}) {
    this._method = method;
    this._uri = uri;
    this._handler = handler;
    this._directory = directory;
  }

  getMethod() {
    return this._method;
  }

  setMethod(method) {
    this._method = method;
    return this;
  }

  getUri() {
    return this._uri;
  }

  setUri(uri) {
    this._uri = uri;
    return this;
  }

  getHandler() {
    return this._handler;
  }

  setHandler(handler) {
    this._handler = handler;
    return this;
  }

  getDirectory() {
    return this._directory;
  }

  setDirectory(directory) {
    this._directory = directory
    return this;
  }
}

Route.GET = GET;
Route.POST = POST;
Route.PUT = PUT;
Route.DELETE = DELETE;
Route.STATIC = STATIC;
Route.ALL = ALL;

module.exports = Route;

File: tools/wptserve/wptserve/wave/data/route.py
class Route:
    def __init__(self, uri="/", handler=None, dependencies={}):
        self._uri = uri
        self._dependencies = dependencies
        self._handler = handler
    
    def get_uri(self):
        return self._uri

    def set_uri(self, uri):
        self._uri = uri
        return self

    def get_handler(self):
        return self._handler

    def set_handler(self, handler):
        self._handler = handler
        return self

    def get_dependencies(self):
        return self._dependencies

    def set_dependencies(self, dependencies):
        self._dependencies = dependencies
        return self